flocker/node/_loop.py

```diff
@@ -404,15 +405,19 @@ class AgentLoopService(object, MultiService):
     :ivar factory: The factory used to connect to the control service.
     """
 
-    def __init__(self):
+    def __init__(self, context_factory):
+        """
+        :param context_factory: TLS context factory for the AMP client.
+        """
         MultiService.__init__(self)
         convergence_loop = build_convergence_loop_fsm(
             self.reactor, self.deployer
         )
         self.logger = convergence_loop.logger
         self.cluster_status = build_cluster_status_fsm(convergence_loop)
-        self.factory = ReconnectingClientFactory.forProtocol(
-            lambda: AgentAMP(self))
```

Comment:
You could keep a direct reference to this factory around and avoid having to insert `wrappedFactory` in a ton of places.